app/services/solidus_square/webhooks/handlers/order_updated.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module SolidusSquare
+  module Webhooks
+    module Handlers
+      class OrderUpdated < Base
+        def call
+          return if not_completed?
+
+          ::Spree::Payment.create!(payment_params) unless order.complete?
+          order.update!(state: "complete")
```

Comment:
This requires also to set completed_at, payment_state(paid) and shipment_state(ready)

Reply:
Can you also update the order.email with the email provided by square?